### PR TITLE
Fixes for sampling update

### PR DESCRIPTION
**Pull request by katxiao**
Fix the following sampling issues
- Create the output file at the beginning of sampling run, regardless of if there are successfully sampled rows
- Add a way to disable writing to output file for internal use
- Only display the progress bar if there are multiple batches or multiple conditions & retries
- Update error messages and error handling

### CODE REVIEW

File: sdv/tabular/utils.py [hunk 1]
```diff
@@ -133,3 +134,47 @@ def unflatten_dict(flat):
...
+        output_file_path (str):
+            The output file path.
+        sampling_error:
+            The error to raise.
+
+    Side Effects:
+        The error will be raised.
+    """
+    if is_tmp_file:
+        print('Error: Sampling terminated. Partial results are stored in a temporary file: '
```
Comment (amontanez24) [inline sdv/tabular/utils.py:173]: Why do we print instead of making it part of the error message?

File: sdv/tabular/base.py [hunk 2]
```diff
@@ -712,16 +739,21 @@ def sample_remaining_columns(self, known_columns, max_tries=100, batch_size_per_
...
-        except (Exception, KeyboardInterrupt) as e:
-            print('Error: Sampling terminated. Partial results are stored in a temporary file: '
-                  f'{output_file_path}. This file will be overridden the next time you sample. '
-                  'Please rename the file if you wish to save these results.')
-            raise e
+            if len(known_columns) == 1 and max_tries == 1:
+                sampled = _sample_function()
+            else:
+                sampled = progress_bar_wrapper(
+                    _sample_function, len(known_columns), 'Sampling remaining columns')
```
Comment (amontanez24) [inline sdv/tabular/base.py:753]: Can we add some unit tests to make sure this logic of only making the progress bar under certain conditions is working?

File: sdv/tabular/utils.py [hunk 3]
```diff
@@ -133,3 +134,47 @@ def unflatten_dict(flat):
             unflattened[key] = value
 
     return unflattened
+
+
+def progress_bar_wrapper(function, pb_total, pb_description):
```
Comment (amontanez24) [inline sdv/tabular/utils.py:139]: Can we add some very basic unit tests for these util functions?

**Review by amontanez24 — COMMENTED** [review]
Looks good! Just a question and some comments about testing

File: sdv/tabular/utils.py (hunk 1 above)
Comment (katxiao) [reply sdv/tabular/utils.py:173]: We wanted to make it print before the actual error is thrown, versus it being a stacked error.

File: sdv/tabular/base.py (hunk 2 above)
Comment (katxiao) [reply sdv/tabular/base.py:753]: Added asserts to the existing unit tests!

**Review by amontanez24 — APPROVED** [review]

**Pull request merged**